File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        Map<String, Integer> map = new HashMap<>();
        
        for (String c : completion) {
            map.put(c, map.getOrDefault(c, 0) + 1);
        }
        
        for (String p : participant) {
            if (map.containsKey(p)) {
                if (map.get(p) == 0) {
                    answer = p;
                    break;
                } else {
                    map.put(p, map.getOrDefault(p, 0) - 1);
                }
            } else {
                answer = p;
                break;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import defaultdict

def solution(participant, completion):
    answer = ''
    map = defaultdict(lambda : 0);
    for name in completion:
        map[name] += 1;
        
    for name in participant:
        if name in map:
            if map[name] == 0:
                answer = name
            else:
                map[name] -= 1
        
        else:
            answer = name
            
        if answer != '':
            break
    return answer